### PR TITLE
Patents

### PR DESCRIPTION
**Pull request by dfdazac**
Adds support for the `snap-patents` dataset introduced in [Large Scale Learning on Non-Homophilous Graphs: New Benchmarks and Strong Simple Methods](https://arxiv.org/abs/2110.14446).

Since this dataset doesn't come with a standard split, experiments are done by training and testing on *n* random splits, and results are reported as the average performance on the test set.
Model selection is done by fixing the random split and tuning hyperparameters on the validation set. To implement this, there's a new flag called `seed` added to `main.py` that allows fixing the random split. This way, we can do model selection by always setting e.g. `seed=0`.

### CODE REVIEW

**Comment by dfdazac** [comment]
Also added a small fix to avoid unnecessarily training the GflowNet when random sampling is used.
